### PR TITLE
Capitalize first letter of all attribute class names

### PR DESCRIPTION
**Pull request by berquist**
Exactly what it says on the tin.  I think we should do this for safety reasons, so there's no confusion about having both a variable and a class name called `scfenergies`, for example.

We can also use this opportunity to change more than just the first letter for some of the more awkward names, but I don't know what makes sense, so I'd like some input.

### CODE REVIEW

File: cclib/attributes/attribute.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def __init__(self, *args, **kwargs):
         )
 
 
-class atombasis(_Attribute):
+class Atombasis(_Attribute):
```
Comment (berquist) [inline cclib/attributes/attribute.py:82]: `AtomBasis`?

File: cclib/attributes/attribute.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def typecheck(self):
             )
 
 
-class aonames(_Attribute):
+class Aonames(_Attribute):
```
Comment (berquist) [inline cclib/attributes/attribute.py:64]: `AoNames`? `AONames`?

File: cclib/attributes/attribute.py [hunk 3]
```diff
@@ -448,7 +448,7 @@ def __init__(self, *args, **kwargs):
         )
 
 
-class nbasis(_Attribute):
+class Nbasis(_Attribute):
```
Comment (berquist) [inline cclib/attributes/attribute.py:451]: `NBasis`?

**Review by berquist — COMMENTED** [review]
I'm personally ok with leaving things as just the first letter capitalized, but so we can see what a few of them might look like...

File: cclib/attributes/attribute.py (hunk 2 above)
Comment (amandadumi) [reply cclib/attributes/attribute.py:64]: I like that first letter leaves no real question and can see this being easy to remember as developing. Obviously AONames makes the most  sense in terms of being clear, so open to that too.

**Review by amandadumi — COMMENTED** [review]
Looks good! I agree with the change right now being first letter only. My heart is split between wanting consistency between classes and the more sensible capitalization options you presented. 

I was wondering if the class capitalization is needed to access the parsed attribute from the user perspective?

**Review by amandadumi — APPROVED** [review]

**Comment by berquist** [comment]
I do also prefer just the first letter.  Anything else will be difficult to make consistent and have everyone be happy.

> I was wondering if the class capitalization is needed to access the parsed attribute from the user perspective?

I don't know what this looks like yet, but we can dynamically change this if need be by introspecting the class name.

**Comment by berquist** [comment]
Waiting to hear from @shivupa and @langner since this is a major change.

I do expect that users will be exposed to this in some way.

**Comment by shivupa** [comment]
> Waiting to hear from @shivupa and @langner since this is a major change.
> 
> I do expect that users will be exposed to this in some way.

Sorry for the delay. I was out of the office for a bit and swimming under work for the other bit... 
I also really like just adopting the capital first letter as a rule even though some cases like `AONames` would warrant other capitalization schemes. Capital first letter lets us just not think about it.

**Pull request merged**